### ISSUE TITLE
[Issue]: Unable to debug - IOError: SQLitePCL.pretty.SQLiteException: disk I/O error

### ISSUE THREAD

**Issue by kaysond** [jellyfin/jellyfin#7971]
### Please describe your bug

After I leave jellyfin running for some time (anywhere from a day to a week), it stops working. I'm unable to get any media to load in the browser. When this happens, the log shows tons of SQLite IO errors which happen directly as a result of a page request. As soon as I restart the container, the problem goes away for a while.

I've checked everything I can think of to make sure jellyfin can access the sqlite dbs:

- the sqlite dbs exist in the container 
- permissions are fine (they're all `rw`)
- none of the dbs are corrupted (ran integrity checks)
- storage is nowhere near full

I'm trying to get some more info, like which sqlite db it's trying to access, and what particular I/O issue there is. I've set up the debug logging as mentioned in the docs, but I'm not getting any debug entries:

```
{
    "Serilog": {
        "MinimumLevel": {
            "Default": "Debug"
        },
        "WriteTo": [
            {
                "Name": "Console",
                "Args": {
                    "outputTemplate": "[{Timestamp:HH:mm:ss}] [{Level:u3}] [{ThreadId}] {SourceContext}: {Message:lj}{NewLine}{Exception}"
                }
            },
            {
                "Name": "Async",
                "Args": {
                    "configure": [
                        {
                            "Name": "File",
                            "Args": {
                                "path": "%JELLYFIN_LOG_DIR%//log_.log",
                                "rollingInterval": "Day",
                                "retainedFileCountLimit": 3,
                                "rollOnFileSizeLimit": true,
                                "fileSizeLimitBytes": 100000000,
                                "outputTemplate": "[{Timestamp:yyyy-MM-dd HH:mm:ss.fff zzz}] [{Level:u3}] [{ThreadId}] {SourceContext}: {Message}{NewLine}{Exception}"
                            }
                        }
                    ]
                }
            }
        ],
        "Enrich": [ "FromLogContext", "WithThreadId" ]
    }
}
```

### Jellyfin Version

10.7.7

### Environment

```markdown
- OS: Debian 11
- Virtualization: Docker
- Clients: N/A
- Browser: N/A
- FFmpeg Version: 4.4.1-Jellyfin
- Playback Method: N/A
- Hardware Acceleration: NVENC
- Plugins: Anime
- Reverse Proxy:apache
- Base URL: none
- Networking: docker
- Storage: NFS
```

### Jellyfin logs

```shell
[14:31:06] [ERR] [164] Jellyfin.Server.Middleware.ExceptionMiddleware: Error processing request. URL GET /Users/c2c136bc1a664f6a86f3529626c7d9a9/Views.
IOError: SQLitePCL.pretty.SQLiteException: disk I/O error
   at SQLitePCL.pretty.SQLiteException.Throw(Int32 rc, Int32 extended, String msg)
   at SQLitePCL.pretty.SQLiteException.CheckOk(sqlite3 db, Int32 rc)
   at SQLitePCL.pretty.StatementImpl.MoveNext()
   at Emby.Server.Implementations.Data.SqliteExtensions.ExecuteQuery(IStatement statement)+MoveNext()
   at Emby.Server.Implementations.Data.SqliteItemRepository.GetItemList(InternalItemsQuery query)
   at Emby.Server.Implementations.Playlists.PlaylistsFolder.GetEligibleChildrenForRecursiveChildren(User user)
   at MediaBrowser.Controller.Entities.Folder.AddChildren(User user, Boolean includeLinkedChildren, Dictionary`2 result, Boolean recursive, InternalItemsQuery query)
   at MediaBrowser.Controller.Entities.Folder.GetChildren(User user, Boolean includeLinkedChildren, InternalItemsQuery query)
   at Emby.Server.Implementations.Playlists.PlaylistsFolder.IsVisible(User user)
   at MediaBrowser.Controller.Entities.Folder.AddChildren(User user, Boolean includeLinkedChildren, Dictionary`2 result, Boolean recursive, InternalItemsQuery query)
   at MediaBrowser.Controller.Entities.Folder.GetChildren(User user, Boolean includeLinkedChildren, InternalItemsQuery query)
   at Emby.Server.Implementations.Library.UserViewManager.GetUserViews(UserViewQuery query)
   at Jellyfin.Api.Controllers.UserViewsController.GetUserViews(Guid userId, Nullable`1 includeExternalContent, String[] presetViews, Boolean includeHidden)
   at lambda_method947(Closure , Object , Object[] )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.SyncObjectResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeActionMethodAsync()
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeNextActionFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|24_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeFilterPipelineAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Jellyfin.Server.Middleware.ServerStartupMessageMiddleware.Invoke(HttpContext httpContext, IServerApplicationHost serverApplicationHost, ILocalizationManager localizationManager)
   at Jellyfin.Server.Middleware.WebSocketHandlerMiddleware.Invoke(HttpContext httpContext, IWebSocketManager webSocketManager)
   at Jellyfin.Server.Middleware.IpBasedAccessValidationMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager)
   at Jellyfin.Server.Middleware.LanFilteringMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager, IServerConfigurationManager serverConfigurationManager)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.ReDoc.ReDocMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Jellyfin.Server.Middleware.RobotsRedirectionMiddleware.Invoke(HttpContext httpContext)
   at Jellyfin.Server.Middleware.LegacyEmbyRouteRewriteMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.ResponseCompression.ResponseCompressionMiddleware.Invoke(HttpContext context)
   at Jellyfin.Server.Middleware.ResponseTimeMiddleware.Invoke(HttpContext context)
   at Jellyfin.Server.Middleware.ExceptionMiddleware.Invoke(HttpContext context)
[14:31:06] [ERR] [79] Jellyfin.Server.Middleware.ExceptionMiddleware: Error processing request. URL GET /Users/c2c136bc1a664f6a86f3529626c7d9a9/Views.
IOError: SQLitePCL.pretty.SQLiteException: disk I/O error
   at SQLitePCL.pretty.SQLiteException.Throw(Int32 rc, Int32 extended, String msg)
   at SQLitePCL.pretty.SQLiteException.CheckOk(sqlite3 db, Int32 rc)
   at SQLitePCL.pretty.StatementImpl.MoveNext()
   at Emby.Server.Implementations.Data.SqliteExtensions.ExecuteQuery(IStatement statement)+MoveNext()
   at Emby.Server.Implementations.Data.SqliteItemRepository.GetItemList(InternalItemsQuery query)
   at Emby.Server.Implementations.Playlists.PlaylistsFolder.GetEligibleChildrenForRecursiveChildren(User user)
   at MediaBrowser.Controller.Entities.Folder.AddChildren(User user, Boolean includeLinkedChildren, Dictionary`2 result, Boolean recursive, InternalItemsQuery query)
   at MediaBrowser.Controller.Entities.Folder.GetChildren(User user, Boolean includeLinkedChildren, InternalItemsQuery query)
   at Emby.Server.Implementations.Playlists.PlaylistsFolder.IsVisible(User user)
   at MediaBrowser.Controller.Entities.Folder.AddChildren(User user, Boolean includeLinkedChildren, Dictionary`2 result, Boolean recursive, InternalItemsQuery query)
   at MediaBrowser.Controller.Entities.Folder.GetChildren(User user, Boolean includeLinkedChildren, InternalItemsQuery query)
   at Emby.Server.Implementations.Library.UserViewManager.GetUserViews(UserViewQuery query)
   at Jellyfin.Api.Controllers.UserViewsController.GetUserViews(Guid userId, Nullable`1 includeExternalContent, String[] presetViews, Boolean includeHidden)
   at lambda_method947(Closure , Object , Object[] )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.SyncObjectResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeActionMethodAsync()
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeNextActionFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|24_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeFilterPipelineAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Jellyfin.Server.Middleware.ServerStartupMessageMiddleware.Invoke(HttpContext httpContext, IServerApplicationHost serverApplicationHost, ILocalizationManager localizationManager)
   at Jellyfin.Server.Middleware.WebSocketHandlerMiddleware.Invoke(HttpContext httpContext, IWebSocketManager webSocketManager)
   at Jellyfin.Server.Middleware.IpBasedAccessValidationMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager)
   at Jellyfin.Server.Middleware.LanFilteringMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager, IServerConfigurationManager serverConfigurationManager)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.ReDoc.ReDocMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Jellyfin.Server.Middleware.RobotsRedirectionMiddleware.Invoke(HttpContext httpContext)
   at Jellyfin.Server.Middleware.LegacyEmbyRouteRewriteMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.ResponseCompression.ResponseCompressionMiddleware.Invoke(HttpContext context)
   at Jellyfin.Server.Middleware.ResponseTimeMiddleware.Invoke(HttpContext context)
   at Jellyfin.Server.Middleware.ExceptionMiddleware.Invoke(HttpContext context)
[14:32:00] [ERR] [78] MediaBrowser.Providers.Folders.FolderMetadataService: Error in Dynamic Image Provider
IOError: SQLitePCL.pretty.SQLiteException: disk I/O error
   at SQLitePCL.pretty.SQLiteException.Throw(Int32 rc, Int32 extended, String msg)
   at SQLitePCL.pretty.SQLiteException.CheckOk(sqlite3 db, Int32 rc)
   at SQLitePCL.pretty.StatementImpl.MoveNext()
   at Emby.Server.Implementations.Data.SqliteExtensions.ExecuteQuery(IStatement statement)+MoveNext()
   at Emby.Server.Implementations.Data.SqliteItemRepository.GetItemList(InternalItemsQuery query)
   at Emby.Server.Implementations.Images.BaseDynamicImageProvider`1.FetchAsync(BaseItem item, ImageType imageType, MetadataRefreshOptions options, CancellationToken cancellationToken)
   at Emby.Server.Implementations.Images.BaseDynamicImageProvider`1.FetchAsync(T item, MetadataRefreshOptions options, CancellationToken cancellationToken)
   at MediaBrowser.Providers.Manager.MetadataService`2.RunCustomProvider(ICustomMetadataProvider`1 provider, TItemType item, String logName, MetadataRefreshOptions options, RefreshResult refreshResult, CancellationToken cancellationToken)
[14:32:00] [ERR] [78] Emby.Server.Implementations.ScheduledTasks.TaskManager: Error
IOError: SQLitePCL.pretty.SQLiteException: disk I/O error
   at SQLitePCL.pretty.SQLiteException.Throw(Int32 rc, Int32 extended, String msg)
   at SQLitePCL.pretty.SQLiteException.CheckOk(sqlite3 db, Int32 rc)
   at SQLitePCL.pretty.StatementImpl.MoveNext()
   at Emby.Server.Implementations.Data.SqliteExtensions.ExecuteQuery(IStatement statement)+MoveNext()
   at Emby.Server.Implementations.Data.SqliteItemRepository.GetItemList(InternalItemsQuery query)
   at MediaBrowser.Controller.Entities.AggregateFolder.LoadChildren()
   at Emby.Server.Implementations.IO.LibraryMonitor.Start()
   at Emby.Server.Implementations.Library.LibraryManager.ValidateMediaLibraryInternal(IProgress`1 progress, CancellationToken cancellationToken)
   at Emby.Server.Implementations.ScheduledTasks.ScheduledTaskWorker.ExecuteInternal(TaskOptions options)
```

### Please attach any screenshots here

![image](https://user-images.githubusercontent.com/1147328/174502026-0929e434-4a8c-43d7-8baa-7351561e121c.png)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by zang74** [comment]
Have you run an fsck on or checked the SMART status of your drive lately? A failing drive is the most consistent way to bring up Disk I/O errors.

**Comment by simwai** [comment]
Some error on my machine too, but in complete normal mode without debugging. 

![image](https://user-images.githubusercontent.com/16225108/178135070-a1b1bb78-224e-4647-a041-04901165ade4.png)
https://sqlite.org/forum/info/229e2b5d9329a1ec1ed7a0d79aa11c7510bf2ac9c24e3c1e545afed04e231681

Could that maybe happen here too? I tried to copy the db and rename it, like in the link above suggested, and that actually worked.

**Comment by kaysond** [comment]
> Could that maybe happen here too? I tried to copy the db and rename it, like in the link above suggested, and that actually worked.
@simwai which db? Do you mean make a copy, then rename both so that jellyfin uses the new copy?
I definitely don't have any hardware problems and my dbs all pass integrity checks...

**Comment by simwai** [comment]
I have no hardware problems too. I get the same exception you mentioned here , sometimes when I initially start Jellyfin with the tray tool. It happened last time today after reboot. I renamed then jellyfin.db and the other db file in the same directory to db.old and copied them and gave the new files the name of the old. Then I started again with tray and somehos the error was gone. I run a 2 or 3 weeks old jellyfin version with some standard plugins installed and no debug mode on.
